app/src/main/java/com/android/bookswap/model/chat/NotificationActionReceiver.kt

```diff
@@ -0,0 +1,33 @@
+package com.android.bookswap.model.chat
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.widget.Toast
+
+/** BroadcastReceiver to handle notification actions. */
+class NotificationActionReceiver : BroadcastReceiver() {
+  /**
+   * Called when the BroadcastReceiver is receiving an Intent broadcast.
+   *
+   * @param context the Context in which the receiver is running.
+   * @param intent the Intent being received.
+   */
+  override fun onReceive(context: Context, intent: Intent) {
+    when (intent.action) {
+      "ACTION_ACCEPT" -> {
+        // Handle accept action
+        requestCode(context, "Accepted")
+      }
+      "ACTION_DECLINE" -> {
+        // Handle decline action
+        requestCode(context, "Declined")
+      }
+    }
+  }
+
+  private fun requestCode(context: Context, action: String) {
```

Comment:
I think the naming is a bit strange, it is misleading as it doesn’t return or handle request codes but shows a Toast.
The name could be changed to enhance understanding of the code.